health_check/backends.py

```diff
@@ -11,6 +11,9 @@
 class BaseHealthCheckBackend:
     def __init__(self):
         self.errors = []
+        self._identifier = getattr(self, '_identifier', self.__class__.__name__)
+        self.critical = getattr(self, 'critical', True)
```

Comment:
Please just use a class attribute. There is not need to overwrite if on each instance. On the contrary, this adds unwanted side effects.
You can no longer overwrite the class attribute on an instance because it's shadowed. That that someone should do that, but it changes behavior.

Besides I don't see the relation to the feature you want to add.